lib/service/Service.ts

```diff
@@ -51,9 +52,13 @@ const argChecks = {
   },
   NON_ZERO_QUANTITY: ({ quantity }: { quantity: number }) => { if (quantity === 0) throw errors.INVALID_ARGUMENT('quantity must not equal 0'); },
   PRICE_NON_NEGATIVE: ({ price }: { price: number }) => { if (price < 0) throw errors.INVALID_ARGUMENT('price cannot be negative'); },
-  SUPPORTED_CURRENCY: ({ currency }: { currency: string }) => {
-    const supportedCurrencies = ['BTC', 'LTC']; // TODO: detect supported currencies automatically instead of hardcoding
-    if (!supportedCurrencies.includes(currency)) throw errors.INVALID_ARGUMENT(`currency ${currency} is not supported`);
+  VALID_CURRENCY: ({ currency }: { currency: string }) => {
+    if (currency.length < 2 || currency.length > 4 || !currency.match(/^[A-Z]+$/))  {
```

Comment:
I have to annoy you again but I just noticed that there are tokens with symbols that consist of 5 characters: `STEEM`, `WAVES`, `STRAT`

What about getting rid of the upper limit entirely?